include/multipass/mount_handler.h

```diff
@@ -18,58 +18,68 @@
 #ifndef MULTIPASS_MOUNT_HANDLER_H
 #define MULTIPASS_MOUNT_HANDLER_H
 
-#include <multipass/disabled_copy_move.h>
-#include <multipass/id_mappings.h>
+#include <multipass/file_ops.h>
 #include <multipass/rpc/multipass.grpc.pb.h>
 #include <multipass/ssh/ssh_key_provider.h>
+#include <multipass/vm_mount.h>
 
 #include <chrono>
-#include <memory>
-#include <string>
 #include <variant>
 
 namespace multipass
 {
 class VirtualMachine;
-struct VMMount;
 
 using ServerVariant = std::variant<grpc::ServerReaderWriterInterface<StartReply, StartRequest>*,
                                    grpc::ServerReaderWriterInterface<LaunchReply, LaunchRequest>*,
                                    grpc::ServerReaderWriterInterface<MountReply, MountRequest>*,
                                    grpc::ServerReaderWriterInterface<RestartReply, RestartRequest>*>;
 
-class MountHandler : private DisabledCopyMove
+struct MountHandler : private DisabledCopyMove
 {
-public:
     using UPtr = std::unique_ptr<MountHandler>;
 
     virtual ~MountHandler() = default;
 
-    // Used to set up anything host side related
-    virtual void init_mount(VirtualMachine* vm, const std::string& target_path, const VMMount& vm_mount) = 0;
-    virtual void start_mount(VirtualMachine* vm, ServerVariant server, const std::string& target_path,
-                             const std::chrono::milliseconds& timeout = std::chrono::minutes(5)) = 0;
-    virtual void stop_mount(const std::string& instance, const std::string& path) = 0;
-    virtual void stop_all_mounts_for_instance(const std::string& instance) = 0;
-    virtual bool has_instance_already_mounted(const std::string& instance, const std::string& path) const = 0;
+    virtual void start(ServerVariant server, std::chrono::milliseconds timeout = std::chrono::minutes(5)) = 0;
+    virtual void stop() = 0;
 
 protected:
-    MountHandler(const SSHKeyProvider& ssh_key_provider) : ssh_key_provider(&ssh_key_provider){};
+    MountHandler() = default;
```

Comment:
Not entirely sure, so correct me if I'm wrong, but I think I need it for the mock/stub classes. If I don't provide a default constructor, those derived classes will need to call the base constructor which does some initialization and validation, which I don't think are needed for those derived classes.

Reply:
Right, and you can't easily extract that validation code into something mockable because, again, no virtual dispatching in the constructor. I suppose that validation code would have to be extracted into a mockable "validator", that the base class c'tor would receive (the real validator could be statically available as a member of the base class, for implementations to use). 

For now, the default constructor is not used, so I think there is no point providing it. How to approach mocking can be part of the work for tests, so maybe we can defer the decision and remove the constructor until it's needed. If it comes down to tests being a little less "unitary" (because they execute the validation code) as the cost to ensuring no empty handlers at compile time, perhaps that's worth paying(?) Maybe not, being protected, the risk of misuse is not big... We should probably look for prior art in other cases in our code. So yeah, can we defer the decision on mocking?

Reply:
Actually, now that I think better, we won't really need in the future either. The only validation in the constructor comes from a single `MP_FILEOPS.status` call, which is already mockable, so it will be very easy to test. I'll remove the default constructor and I don't think it will need to come back again later.